tests/test_service.py

```diff
@@ -830,3 +830,155 @@ def testGetAlertWithMultipleTriggers(self):
+        self.assertEqual(child_alert.alertType, 'COMPOSITE_CHILD')
+        self.assertTrue(isinstance(child_alert, Alert))
+
+    def testAddTriggerToChildAlert(self):
+        comp_alert = self._createCompAlert()
+
+        with mock.patch('requests.Session.post', return_value=MockResponse(json.dumps(childAlert_1), 200)) as mock_add_childalert:
+            child_alert = self.argus.alerts.add_child_alert_to_composite_alert(comp_alert.id,
+                                                                               Alert.from_dict(childAlert_1))
+            self.assertEqual(child_alert.alertType, 'COMPOSITE_CHILD')
+            self.assertTrue(isinstance(child_alert, Alert))
+            call_args = tuple(mock_add_childalert.call_args)
+            uri_path = os.path.join(endpoint, "alerts/{}/children".format(comp_alert.id))
+            self.assertIn((uri_path,), call_args)
+
+        with mock.patch('requests.Session.post', return_value=MockResponse(json.dumps([childAlert_trigger_1]), 200)) as mock_trigger_post:
+            trigger_obj = Trigger.from_dict(childAlert_trigger_1)
+            delattr(trigger_obj, "id")
+            trigger = child_alert.triggers.add(trigger_obj)
+            self.assertTrue(isinstance(trigger, Trigger))
+            call_args = tuple(mock_trigger_post.call_args)
+            uri_path = os.path.join(endpoint, "alerts/{}/triggers".format(child_alert.id))
+            self.assertIn((uri_path,), call_args)
+
+    def testAddNotification(self):
+        comp_alert = self._createCompAlert()
+
+        with mock.patch('requests.Session.post', return_value=MockResponse(json.dumps([compAlert_notification]), 200)) as mock_notification:
+            notification_obj = Notification.from_dict(compAlert_notification)
+            delattr(notification_obj, "id")
+            notification = comp_alert.notifications.add(notification_obj)
+            self.assertTrue(isinstance(notification, Notification))
+            call_args = mock_notification.call_args
+            uri_path = os.path.join(endpoint, "alerts/{}/notifications".format(comp_alert.id))
+            self.assertIn((uri_path,), call_args)
+
+
+    def testDeleteChildAlert(self):
+        comp_alert = self._createCompAlert()
+        with mock.patch('requests.Session.post', return_value=MockResponse(json.dumps(childAlert_1), 200)):
+            child_alert = self.argus.alerts.add_child_alert_to_composite_alert(comp_alert.id, Alert.from_dict(childAlert_1))
+            self.assertEqual(child_alert.alertType, 'COMPOSITE_CHILD')
+            self.assertTrue(isinstance(child_alert, Alert))
+
+        ''' Before delete we can get the object for child_alert.id '''
+        res = self.argus.alerts.get(child_alert.id)
+        with mock.patch('requests.Session.delete', return_value=MockResponse("", 200)) as mock_delete:
+            self.argus.alerts.delete_child_alert_from_composite_alert(comp_alert.id, child_alert.id)
```

Comment:
Where is the check to make sure that the object is deleted from the local cache, the [code at line 683](https://github.com/salesforce/python-argusclient/pull/28/files#diff-16d091ec55cf0b28f5e8e4b8bbebf2fcc61009d9c14c35b83b4e63cb10c729a3R683) where you fixed a typo?

Reply:
As part of the testDeleteChildAlert we perform the following steps
1. Create composite alert (Line 901), where the child_alert_id will be added to self._coll
2. I retrieve the object in line 906 (This ensures that the child alert object is present in self_coll)
3. Line 908 I am deleting the child_alert_id, this would remove the child_alert_id from self._coll
4. Line 915 we are mocking requests.get to return 404, which would return/raise ArgusObjectNotFoundException only when child_alert_id is not present in self._coll
Hope that is clear.  If not we can get on a call and discuss what exactly is happening.

Reply:
Thanks a lot for the explanation, it makes more sense to me. I guess I can suggest a slight improvement to the comment.